code/modules/mob/living/carbon/xenomorph/castes/defender/abilities_defender.dm

```diff
@@ -396,17 +396,21 @@
 
 /datum/action/xeno_action/activable/centrifugal_force/action_activate()
 	if(spin_loop_timer)
-		deltimer(spin_loop_timer)
-		spin_loop_timer = null
+		stop_spin()
 		return
+	if(!can_use_action(TRUE))
+		return fail_activate()
 	if(!do_after(owner, 0.5 SECONDS, TRUE, owner, BUSY_ICON_DANGER))
 		return fail_activate()
+	if(!can_use_action(TRUE))
+		return fail_activate()
```

Comment:
can't you add this as an extra check in the do_after?

Reply:
yea